snet_cli/commands.py

```diff
@@ -504,3 +510,239 @@ def transact(self):
+            self._printout("Please provide values to populate your service.json file\n")
+
+        if self.args.name:
+            init_args["name"] = self.args.name
+        elif not accept_all_defaults:
+            init_args["name"] = input('Choose a name for your service: (default: "{}")\n'.format(init_args["name"])) or init_args["name"]
+
+        if self.args.model:
+            init_args["model"] = self.args.model
+        elif not accept_all_defaults:
+            init_args["model"] = input('Choose the path to your service\'s model directory: (default: "{}")\n'.format(init_args["model"])) or init_args["model"]
+
+        if self.args.organization:
+            init_args["organization"] = self.args.organization
+        elif not accept_all_defaults:
+            init_args["organization"] = input('Choose an organization to register your service under: (default: "{}")\n'.format(init_args["organization"])) or init_args["organization"]
+
+        if self.args.price:
+            init_args["price"] = self.args.price
+        elif not accept_all_defaults:
+            init_args["price"] = input('Choose a price in AGI to call your service: (default: {})\n'.format(init_args["price"])) or init_args["price"]
+
+        if self.args.endpoint:
+            init_args["endpoint"] = self.args.endpoint
+        elif not accept_all_defaults:
+            init_args["endpoint"] = input('Endpoint to call the API for your service: (default: "{}")\n'.format(init_args["endpoint"])) or init_args["endpoint"]
+
+        if self.args.tags:
+            init_args["tags"] = self.args.tags
+        elif not accept_all_defaults:
+            tags_user_input = input("Input a list of tags for your service: (default: {})\n".format(init_args["tags"])) or init_args["tags"]
+            if tags_user_input and " " in tags_user_input and "," in tags_user_input:
+                raise ValueError("Please either provide a list separated either by comma or space, not both")
+            elif tags_user_input and "," in tags_user_input:
+                init_args["tags"] = tags_user_input.split(",")
+            elif tags_user_input and " " in tags_user_input:
+                init_args["tags"] = tags_user_input.split()
+            for idx, tag in enumerate(init_args["tags"]):
+                init_args["tags"][idx] = tag.strip()
+
+        if self.args.description:
+            init_args["metadata"]["description"] = self.args.description
+        elif not accept_all_defaults:
+            init_args["metadata"]["description"] = input('Input a description for your service: (default: "{}")\n'.format(init_args["metadata"]["description"])) or init_args["metadata"]["description"]
+
+        with open("service.json", "w") as f:
+            json.dump(init_args, f, indent=4, ensure_ascii=False)
+        self._printout(json.dumps(init_args, indent=4, sort_keys=True))
+        self._printout("\nservice.json file has been created!")
+
+    def publish(self):
+        self._get_network()
```

Comment:
If I re-run publish after already publishing, this command appears to go through the whole process again (creating a new Agent contract, etc). Since update checks to make sure the service has been published on the specified network, should publish make sure it hasn't?

Reply:
Yea for the moment I would check if the service has already been published and spit out a message like "this service has already been published to this network", as we said that if the user wants to re-publish the service he should basically use a new service.json file (either passing it explicitly via --config, or have a new directory); in the next iteration I would reupload model/metadata to IPFS, fetch the updatable fields from the blockchain (endpoint, price, metadata, etc), compare each field with what's in the service.json file (and the metadataURI with the hash of the newly uploaded metadata) and prompt the user to update whatever's changed, but that seems too big of an item to go in this CR process.